### PR TITLE
[ENH] scitype variable name and doc string purge

### PR DESCRIPTION
**Pull request by TonyBagnall**
#### Reference Issues/PRs
 dependent on 

part of #42 inspired by #709 and dependent on #709 

#### What does this implement/fix? Explain your changes.

This is a manual refactoring of the registry references to the term "scitype". The scitype set up is basically to have a string identifier associated with base classes.  Why this is needed is beyond me, seems to be a kind of implicit string based inheritance hierarchy. However, its here, so to begin with I will refactor, much of it manually because IDE python refactoring is poor. There is no functionality changing here at all, its purely cosmetic. One of the most confusing things about scitype is its used for estimators and data types. So scitype == Panel refers to a data structure, scitype == classifier to a class implementing BaseClassifier. Lets not do that overloading any more.

The main change is to replace "scitype" with "identifier". sctiypes are defined gere

BASE_CLASS_REGISTER = [
    ("object", BaseObject, "object"),
    ("estimator", BaseEstimator, "estimator = object with fit"),
    ("classifier", BaseClassifier, "time series classifier"),
    ("clusterer", BaseClusterer, "time series clusterer"),
    ("early_classifier", BaseEarlyClassifier, "early time series classifier"),
    ("forecaster", BaseForecaster, "forecaster"),
    ("metric", BaseMetric, "performance metric"),
    ("network", BaseDeepNetwork, "deep learning network"),
    ("regressor", BaseRegressor, "time series regressor"),
    ("series-annotator", BaseSeriesAnnotator, "time series annotator"),
    ("splitter", BaseSplitter, "time series splitter"),
    ("transformer", BaseTransformer, "time series transformer"),
    (
        "collection-transformer",
        BaseCollectionTransformer,
        "time series collection transformer",
    ),
]

BASE_CLASS_SCITYPE_LIST = pd.DataFrame(BASE_CLASS_REGISTER)[0].tolist()

becomes

BASE_CLASS_IDENTIFIER_LIST = pd.DataFrame(BASE_CLASS_REGISTER)[0].tolist()

etc etc.

### CODE REVIEW

File: aeon/utils/_testing/scenarios_getter.py [hunk 1]
```diff
@@ -90,20 +88,6 @@ def retrieve_scenarios(obj, filter_tags=None):
...
-    if not isclass(obj):
-        obj = type(obj)
-    if not isinstance(obj, tuple(BASE_CLASS_LIST)):
-        raise TypeError("obj must be instance of an aeon base class, or a base class")
-
-    for i in range(len(BASE_CLASS_SCITYPE_LIST)):
-        if isinstance(obj, BASE_CLASS_LIST[i]):
-            return BASE_CLASS_SCITYPE_LIST[i]
-
-
```
Comment (MatthewMiddlehurst) [inline aeon/utils/_testing/scenarios_getter.py:106]: Is this not used anywhere?

**Review by MatthewMiddlehurst — APPROVED** [review]
Fine if comment below is resolved.

File: aeon/utils/_testing/scenarios_getter.py (hunk 1 above)
Comment (TonyBagnall) [reply aeon/utils/_testing/scenarios_getter.py:106]: nope

**Comment by kevinlu1248** [comment]
@TonyBagnall Would you like me to write a Sweep Rule for this?

**Comment by MatthewMiddlehurst** [comment]
I think it may be a bit too complex and convoluted to be honest even as a person 🙂. The term is used in many different contexts for some reason. Probably better as a gradual change.

**Comment by TonyBagnall** [comment]
thanks for the idea, but its deeply embedded all over the place and touches hundreds off files

**Pull request merged**